packages/custom_lint/lib/src/workspace.dart

```diff
@@ -503,24 +505,28 @@ class CustomLintProject {
   ) async {
     final directory = Directory(contextRoot.root);
 
-    final projectPubspec = await parsePubspec(directory)
-        // ignore: avoid_types_on_closure_parameters
-        .catchError((Object err, StackTrace stack) {
-      throw PubspecParseError._(
-        directory.path,
-        error: err,
-        errorStackTrace: stack,
-      );
-    });
-    final projectPackageConfig = await parsePackageConfig(directory)
-        // ignore: avoid_types_on_closure_parameters
-        .catchError((Object err, StackTrace stack) {
-      throw PackageConfigParseError._(
-        directory.path,
-        error: err,
-        errorStackTrace: stack,
-      );
-    });
+    final projectPubspec = await parsePubspec(directory).catchError(
+      // ignore: avoid_types_on_closure_parameters
+      (Object err, StackTrace stack) {
+        throw PubspecParseError._(
+          directory.path,
+          error: err,
+          errorStackTrace: stack,
+        );
+      },
+      test: (e) => e is! MissingFileError,
```

Comment:
Here and below we only catch parse errors, passing through missing file errors uncaught.